flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/fs/ForStFlinkFileSystem.java

```diff
@@ -273,18 +277,38 @@ public synchronized boolean exists(final Path f) throws IOException {
         }
 
         if (FileOwnershipDecider.shouldAlwaysBeLocal(f)) {
-            return localFS.exists(mappingEntry.getSourcePath())
-                    || delegateFS.exists(mappingEntry.getSourcePath());
+            return localFS.exists(mappingEntry.getSourcePath());
         } else {
+            // Should be protected with synchronized, since the file closing is not an atomic
+            // operation, see FSDataOutputStreamWithEntry.close()
+            synchronized (mappingEntry) {
+                if (mappingEntry.isWriting()) {
+                    return true;
+                }
+            }
             return delegateFS.exists(mappingEntry.getSourcePath());
         }
     }
 
     @Override
     public synchronized FileStatus getFileStatus(Path path) throws IOException {
```

Comment:
Does `getFileBlockLocations` need to be modified like `getFileStatus`?

Reply:
No need as it never be called for now. But I still change the `getFileBlockLocations`.